### PR TITLE
Split packages so we can extract `gitops-toolkit`

### PR DESCRIPTION
**Pull request by luxas**

### CODE REVIEW

File: pkg/storage/manifest/storage.go [hunk 1]
```diff
@@ -1,22 +1,40 @@
...
+	return &ManifestStorage{
+		Storage: ss,
+	}, nil
+}
+
+// NewManifestStorage constructs a new storage that watches unstructured manifests in the specified directory,
+// decodable using the given serializer. However, all changes in the manifest directory, are also propagated to
+// the structured data directory that's backed by the default storage implementation. Writes to this storage are
+// propagated to both the manifest directory, and the data directory.
+func NewTwoWayManifestStorage(manifestDir, dataDir string, ser serializer.Serializer) (*ManifestStorage, error) {
```
Comment (twelho) [inline pkg/storage/manifest/storage.go]: `TwoWay` might be a bit confusing as this doesn't propagate data directory changes to the manifest directory (like the name implies). Is `DataManifestStorage` better?

File: pkg/storage/watch/storage.go [hunk 2]
```diff
@@ -42,8 +40,12 @@ func NewGenericWatchStorage(s storage.Storage) (WatchStorage, error) {
 		return nil, err
 	}
 
+	// TODO: Fix this
+	gvs := s.Serializer().Scheme().PreferredVersionAllGroups()
+	groupName := gvs[0].Group
```
Comment (twelho) [inline pkg/storage/watch/storage.go]: OK for now

File: pkg/storage/manifest/storage.go (hunk 1 above)
Comment (luxas) [reply pkg/storage/manifest/storage.go]: It does propagate all the manifest that it knows about from before, so I think this is ok for now at least. We can talk about naming later.

File: pkg/serializer/serializer.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+// EncodeJSON encodes the specified object for a specific version to pretty JSON bytes
+func (s *serializer) EncodeJSON(obj runtime.Object) ([]byte, error) {
+	return s.encode(obj, runtime.ContentTypeJSON, true)
+}
+
+func (s *serializer) encode(obj runtime.Object, mediaType string, pretty bool) ([]byte, error) {
+	info, ok := runtime.SerializerInfoForMediaType(s.codecs.SupportedMediaTypes(), mediaType)
+	if !ok {
+		return nil, fmt.Errorf("unable to locate encoder -- %q is not a supported media type", mediaType)
```
[the remark this answers is not in the document]
Comment (luxas) [reply pkg/serializer/serializer.go]: This was not new code; just the same as before.

**Comment by luxas** [comment]
Merging to get going with the gitops-toolkit extraction

**Pull request merged**